src/navigation/RootStackNavigation.tsx

```diff
@@ -84,104 +98,193 @@ const RootStackNavigation = () => {
+                      />
+                      <RootStack.Screen
+                        name="LoginWithPinScreen"
+                        component={LoginWithPinScreen}
+                        options={{ cardStyleInterpolator: CardStyleInterpolators.forFadeFromCenter }}
+                      />
+                      <RootStack.Screen name="NewWalletSuccessScreen" component={NewWalletSuccessScreen} />
+                      <RootStack.Screen
+                        name="InWalletTabsNavigation"
+                        component={InWalletTabsNavigation}
+                        options={{ cardStyleInterpolator: CardStyleInterpolators.forFadeFromCenter }}
+                      />
+                      <RootStack.Screen name="PinCodeCreationScreen" component={PinCodeCreationScreen} />
+                      <RootStack.Screen name="SettingsScreen" component={SettingsScreen} />
+                      <RootStack.Screen name="NewContactScreen" component={NewContactScreen} />
+                      <RootStack.Screen name="ContactScreen" component={ContactScreen} />
+                      <RootStack.Screen name="NewAddressScreen" component={NewAddressScreen} />
+                      <RootStack.Screen name="EditContactScreen" component={EditContactScreen} />
+                      <RootStack.Screen name="EditAddressScreen" component={EditAddressScreen} />
+                      <RootStack.Screen name="AddressDiscoveryScreen" component={AddressDiscoveryScreen} />
+                      <RootStack.Screen name="NewWalletNameScreen" component={NewWalletNameScreen} />
+                      <RootStack.Screen name="NewWalletIntroScreen" component={NewWalletIntroScreen} />
+                      <RootStack.Screen name="SelectImportMethodScreen" component={SelectImportMethodScreen} />
+                      <RootStack.Screen name="DecryptScannedMnemonicScreen" component={DecryptScannedMnemonicScreen} />
+                      <RootStack.Screen name="ImportWalletSeedScreen" component={ImportWalletSeedScreen} />
+                      <RootStack.Screen name="AddBiometricsScreen" component={AddBiometricsScreen} />
+                      <RootStack.Screen name="EditWalletNameScreen" component={EditWalletNameScreen} />
+                      <RootStack.Screen name="CustomNetworkScreen" component={CustomNetworkScreen} />
+                      <RootStack.Screen
+                        name="ImportWalletAddressDiscoveryScreen"
+                        component={ImportWalletAddressDiscoveryScreen}
+                      />
+                    </RootStack.Group>
+                  </RootStack.Navigator>
+                </WalletConnectContextProvider>
+              </AnalyticsProvider>
+            </NavigationScrollContextProvider>
+          </AppUnlockHandler>
         </NavigationContainer>
       </Host>
     </GestureHandlerRootView>
   )
 }
 
 export default RootStackNavigation
+
+const AppUnlockHandler = ({ children }: { children: ReactNode }) => {
```

Comment:
This code is simply copied from the `App.tsx`, with the only difference that I am now using `navigation.navigate` instead of the custom `navigateRootStack` function.

Reply:
That's great!